server/service/global_policies.go

```diff
@@ -158,19 +169,41 @@ func deleteGlobalPoliciesEndpoint(ctx context.Context, request interface{}, svc
 // DeleteGlobalPolicies deletes the given policies from the database.
 // It also deletes the given ids from the failing policies webhook configuration.
 func (svc Service) DeleteGlobalPolicies(ctx context.Context, ids []uint) ([]uint, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
+	// First check if authorized to read policies
+	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionRead); err != nil {
 		return nil, err
 	}
 	if len(ids) == 0 {
 		return nil, nil
 	}
+	policiesByID, err := svc.ds.PoliciesByID(ctx, ids)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting policies by ID")
+	}
+	// Then check if authorized to write policies
+	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
 	if err := svc.removeGlobalPoliciesFromWebhookConfig(ctx, ids); err != nil {
 		return nil, ctxerr.Wrap(ctx, err, "removing global policies from webhook config")
 	}
-	ids, err := svc.ds.DeleteGlobalPolicies(ctx, ids)
+	deletedIDs, err := svc.ds.DeleteGlobalPolicies(ctx, ids)
```

Comment:
Nothing to change on this PR, I just find odd that `Datastore.DeleteGlobalPolicies`'s MySQL implementation basically returns the given `ids` (probably a bug, but nbd).